### PR TITLE
ci: adds azure GHA test

### PR DESCRIPTION
**Pull request by faiq**
**What problem does this PR solve?**:
Moves azure tests to GHA 

**Which issue(s) does this PR fix?**:

* https://jira.d2iq.com/browse/D2IQ-NUMBER

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -222,7 +222,7 @@ github-token.txt:
 .PHONY: buildx
 buildx:
 buildx:
-	 docker buildx create --driver=docker-container --use --name=konvoy-image-builder || true
+	 docker buildx create --use --name=konvoy-image-builder || true
```
Comment (dkoshkin) [inline Makefile:225]: Why was this change needed?

File: .github/workflows/azure-e2e.yaml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+jobs:
+  rune2e:
+    strategy:
+      fail-fast: false
+      max-parallel: 10
+      matrix:
+        os:
+        - "ubuntu 20.04"
+        buildConfig:
+        - "basic"
```
Comment (dkoshkin) [inline .github/workflows/azure-e2e.yaml:18]: Im loving how easy it is to add new configs to the build matrix!

File: Makefile (hunk 1 above)
Comment (faiq) [reply Makefile:225]: It's not needed, but it makes things a lot faster because the cache can be re-used

Comment (faiq) [reply Makefile:225]: I can pull it out to separate PR if you'd like

File: .github/workflows/azure-e2e.yaml [hunk 3]
```diff
@@ -0,0 +1,52 @@
+'on':
+  pull_request:
+    branches:
+    - '*'
```
Comment (supershal) [inline .github/workflows/azure-e2e.yaml:4]: I think this filter is not needed.

File: magefile.go [hunk 4]
```diff
@@ -230,8 +233,10 @@ func getBuildPath(buildOS, buildInfra string) string {
 	formattedOS = strings.Replace(formattedOS, ".", "", 1)
 
 	formattedOS = strings.Replace(formattedOS, "redhat", "rhel", 1)
-	formattedOS = strings.Replace(formattedOS, "1804", "18", 1)
-	formattedOS = strings.Replace(formattedOS, "2004", "20", 1)
+	if buildInfra == aws {
+		formattedOS = strings.Replace(formattedOS, "1804", "18", 1)
+		formattedOS = strings.Replace(formattedOS, "2004", "20", 1)
+	}
```
Comment (faiq) [inline magefile.go:239]: This is annoying we have different file names for azure and aws

File: .github/workflows/azure-e2e.yaml [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+          ref: ${{ github.ref }}
+
+      - name: Set up Go
+        uses: actions/setup-go@v3
+        with:
+          go-version-file: 'go.mod'
```
Comment (supershal) [inline .github/workflows/azure-e2e.yaml:30]: please use `cache: true` to run this action faster

File: .github/workflows/azure-e2e.yaml [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+
+      - name: Docker Login
+        uses: docker/login-action@v2
+        with:
+          registry: docker.io
+          username: mesosphereci
```
Comment (supershal) [inline .github/workflows/azure-e2e.yaml:36]: can we use org level secrets. `DOCKER_READ_WRITE_USERNAME` and `DOCKER_READ_WRITE_PASSWORD` ?
Assuming that they should be ok to use even it for a public repo.

File: Makefile (hunk 1 above)
Comment (dkoshkin) [reply Makefile:225]: Yes, that would be good.

Comment (supershal) [reply Makefile:225]: This was added when we were trying arm64 support. We later find out that we dont need it. This argument `--driver=docker-container` unnecessary slows builds down and does not cache docker file system layers between builds.

File: .github/workflows/azure-e2e.yaml (hunk 6 above)
Comment (faiq) [reply .github/workflows/azure-e2e.yaml:36]: Those are not org level secrets

File: magefile.go (hunk 4 above)
Comment (supershal) [reply magefile.go:239]: 💯

File: Makefile (hunk 1 above)
Comment (faiq) [reply Makefile]: removed and will file another PR

File: magefile.go (hunk 4 above)
Comment (faiq) [reply magefile.go]: i changed the file names as a part of this PR

File: .github/workflows/azure-e2e.yaml (hunk 6 above)
Comment (faiq) [reply .github/workflows/azure-e2e.yaml]: Used other secrets

**Review by supershal — APPROVED** [review]
lgtm

**Review by dkoshkin — APPROVED** [review]
Lets wait for the new Azure and AWS tests to pass before mergning, but this is great!

**Comment by supershal** [comment]
Test looks good. marked Ubuntu tests GHA as required and removed same tests in TC from the `required` list. The TC jobs are paused as well.
